src/main/java/org/commcare/formplayer/services/FormDefinitionService.java

```diff
@@ -0,0 +1,39 @@
+package org.commcare.formplayer.services;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.commcare.formplayer.exceptions.MultipleFormDefsFoundException;
+import org.commcare.formplayer.objects.FormDefinition;
+import org.commcare.formplayer.repo.FormDefinitionRepo;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheConfig;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@CacheConfig(cacheNames = {"form_definition"})
+public class FormDefinitionService {
+
+    private final Log log = LogFactory.getLog(FormDefinitionService.class);
+
+    @Autowired
+    private FormDefinitionRepo formDefinitionRepo;
+
+
+    @Cacheable
```

Comment:
No need to include the full form def in the cache key:

```suggestion
    @Cacheable(key={ #appId, #appVersion, #formXmlns }
```

Reply:
Was trying to find the write docs to reference how to do this, so thanks for flagging. See e33f2195c55b9d877723166e4b17390067c05145